### PR TITLE
[IotCentral] Update C# sdk, Change error to be CloudException, and remove regex check for IOTC app

### PR DESCRIPTION
**Pull request by kawayiboy**
[IotCentral] Update C# sdk, Change error to be CloudException, and remove regex check for IOTC app

### CODE REVIEW

**Comment by isra-fel** [comment]
Please remove file `sdk/iotcentral/Microsoft.Azure.Management.IotCentral/src/Generated/code-model-v1` and put the link of related swagger review PR in the comments. Thanks!

**Comment by kawayiboy** [comment]
Please wait until March 19th to merge it when all the other sdks are published, Thanks

**Comment by kawayiboy** [comment]
Here is the related swagger change: https://github.com/Azure/azure-rest-api-specs/pull/8395, it's ok to merge now, the sdk have been published, thanks

**Review by isra-fel — APPROVED** [review]

**Pull request merged**
